connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -418,10 +418,10 @@ public boolean startTask(
 
             ClassLoader savedLoader = plugins.currentThreadLoader();
             try {
-                final ConnectorConfig connConfig = new ConnectorConfig(plugins, connProps);
-                String connType = connConfig.getString(ConnectorConfig.CONNECTOR_CLASS_CONFIG);
+                String connType = connProps.get(ConnectorConfig.CONNECTOR_CLASS_CONFIG);
```

Comment:
By the time it gets here, there are other checks that prevent `null` values for this config. That's the reason I didn't add an explicit check here. But I would be open to adding one if you think it would be useful.